From 212db063cbe6dd8b753ba2fa77c716876f5393cc Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@compasses.com>
Date: Mon, 10 Aug 2020 17:22:16 -0500
Subject: [PATCH 19/44] dts: Don't assign clock rates in consumer nodes.

Also continue to take needed clock references using EHCI as a work
around to some consumers not taking a reference to a required clock
input. This undoes changes not related to beta hardware to the former
state.
---
 .../boot/dts/renesas/beacon-renesom-baseboard.dtsi     | 10 ++++------
 arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi    |  4 ----
 2 files changed, 4 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
index 166ce28f8c6c..439a76b80de0 100644
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
@@ -228,8 +228,6 @@
 
 &audio_clk_a {
 	clock-frequency = <24576000>; /* Support 48KHz */
-	assigned-clocks = <&versaclock6_bb 4>;
-	assigned-clock-rates = <24576000>;
 };
 
 &audio_clk_b {
@@ -300,12 +298,14 @@
 &ehci0 {
 	dr_mode = "otg";
 	status = "okay";
-	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>;
+	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 3>, <&versaclock6_bb 4>;
 };
 
 &ehci1 {
 	status = "okay";
-	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>;
+	/* Hack, we are asking the EHCI to take a clk ref to the Ethernet Phy
+	refclock so Ethernet will work */
+	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 4>;
 };
 
 &hdmi0 {
@@ -465,8 +465,6 @@
 		PLLVDD-supply = <&reg_audio>;
 		SPKVDD1-supply = <&reg_audio>;
 		SPKVDD2-supply = <&reg_audio>;
-		assigned-clocks = <&versaclock6_bb 3>;
-		assigned-clock-rates = <24000000>;
 		gpio-cfg = <
 			0x0000 /* 0:Default */
 			0x0000 /* 1:Default */
diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
index 737a0d204e5b..2f2ded41445d 100644
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
@@ -93,8 +93,6 @@
 	pinctrl-names = "default";
 	phy-handle = <&phy0>;
 	phy-mode = "rgmii-txid";
-	assigned-clocks = <&versaclock5 4>;
-	assigned-clock-rates = <125000000>;
 	status = "okay";
 
 	phy0: ethernet-phy@0 {
@@ -132,8 +130,6 @@
 
 &usb_extal_clk {
 	clock-frequency = <50000000>;
-	assigned-clocks = <&versaclock5 3>;
-	assigned-clock-rates = <50000000>;
 };
 
 &usb3s0_clk {
-- 
2.17.1

